tests/test_extensions/test_extensions.py

```diff
@@ -9,6 +9,8 @@ def test_extensions(self):
         self.found_files_eq('ext:c', ['main.c', 'dot_c.c'])
         self.found_files_eq('ext:cpp', ['hello-world.cpp'])
         self.found_files_eq('ext:inc', ['hello-world.inc'])
+        # Make sure that multiple ext filters are joined with 'or' instead of 'and.'
+        self.found_files_eq('ext:c ext:cpp', ['main.c', 'dot_c.c', 'hello-world.cpp'])
```

Comment:
I'd like to see this broken off into a separate test, just to accentuate that it's testing the underlying union_only machinery, not (so much) the ExtFilter itself.